crates/block-producer/src/mempool/mod.rs

```diff
@@ -339,17 +359,62 @@ impl Mempool {
 
         let batches = self.block_in_progress.take().expect("No block in progress to be failed");
 
-        // Remove all transactions from the graphs.
-        let purged = self.batches.remove_batches(batches).expect("Batch should be removed");
-        let transactions = purged.into_values().flatten().collect();
+        // Revert all transactions. This is the nuclear (but simplest) solution.
+        //
+        // We currently don't have a way of determining why this block failed so take the safe route
+        // and just nuke all associated transactions.
+        //
+        // TODO: improve this strategy, e.g. count txn failures (as well as in e.g. batch failures),
+        // and only revert upon exceeding some threshold.
```

Comment:
Let's create an issue to address this in the future (unless we already have one).

Reply:
https://github.com/0xPolygonMiden/miden-node/issues/594